spring/src/main/java/com/linecorp/armeria/spring/ArmeriaAutoConfiguration.java

```diff
@@ -0,0 +1,213 @@
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.spring;
+
+import static com.linecorp.armeria.spring.MetricNames.serviceMetricName;
+
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.thrift.TBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.json.MetricsModule;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.google.common.base.Strings;
+
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponse;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpSessionProtocols;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.Service;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.server.docs.DocServiceBuilder;
+import com.linecorp.armeria.server.http.AbstractHttpService;
+import com.linecorp.armeria.server.http.healthcheck.HealthChecker;
+import com.linecorp.armeria.server.http.healthcheck.HttpHealthCheckService;
+import com.linecorp.armeria.server.logging.DropwizardMetricCollectingService;
+import com.linecorp.armeria.spring.ArmeriaSettings.Port;
+
+import io.netty.util.NetUtil;
+
+@Configuration
+@EnableConfigurationProperties(ArmeriaSettings.class)
+public class ArmeriaAutoConfiguration {
+
+    private static final Logger logger = LoggerFactory.getLogger(ArmeriaAutoConfiguration.class);
+
+    private static final HealthChecker[] EMPTY_HEALTH_CHECKERS = new HealthChecker[0];
+
+    @Configuration
+    @ConditionalOnMissingBean(Server.class)
+    public static class ArmeriaServerConfiguration {
```

Comment:
- I'm curious why this class has to be an inner class of `ArmeriaAutoConfiguration`. Is it some sort of Spring convention?
- How about just `ArmeriaConfiguration`?